### ISSUE TITLE
[Java] AvoidAccessibilityAlteration: What is this rule about? 

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2525]
**Description:**
[AvoidAccessibilityAlteration]
(https://pmd.github.io/latest/pmd_rules_java_errorprone.html#avoidaccessibilityalteration).

Can you explain what's the purpose of the above rule?

Why is the rule restricted only to constructors and not methods and fields?

Why is the rule enforced despite the presence of a PrivilegedAction block? 

Surely, if reflection code is encapsulated within a PrivilegedAction then the accessibility violation is deliberate i.e., the coder intends to do exactly that (break encapsulation).

Spotbugs has a similar rule that enforces use of accessibility code within a doPrivilegedAction block. 

[DP_DO_INSIDE_DO_PRIVILEGED](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dp-method-invoked-that-should-be-only-be-invoked-inside-a-doprivileged-block-dp-do-inside-do-privileged)

I'm more comfortable with the above interpretation of how to handle reflection in code.

Secondly, shouldn't the AvoidAccessibilityAlteration check be only for setAccessible(true) since false forces the JVM to enforce the language semantics?

Adding StackOverflow Q&A for context:

[How to limit setAccessible to only "legitimate" uses](https://stackoverflow.com/questions/2481862/how-to-limit-setaccessible-to-only-legitimate-uses)
